pkg/events/derive/process_execute_failed.go

```diff
@@ -0,0 +1,145 @@
+package derive
+
+import (
+	"fmt"
+	"sync"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+func ProcessExecuteFailed() DeriveFunction {
+	gen, err := newProcessExecuteFailedGeneratorSingleton()
+	if err != nil {
+		logger.Errorw("failed to init derive function for ProcessExecuteFiled", "error", err)
+		return nil
+	}
+	return func(event trace.Event) ([]trace.Event, []error) {
+		var errs []error
+		var derivedEvents []trace.Event
+		derivedEvent, err := gen.deriveEvent(&event)
+		if err != nil {
+			errs = append(errs, err)
+		}
+		if derivedEvent != nil {
+			derivedEvents = append(derivedEvents, *derivedEvent)
+		}
+		return derivedEvents, errs
+	}
+}
+
+// execEndInfo stores information about the end of process execution operation.
+type execEndInfo struct {
+	returnCode int
+	timestamp  int
+}
+
+// execFailedGenerator is the object which implement the ProcessExecuteFailed event derivation
+type execFailedGenerator struct {
+	execEndInfo *lru.Cache[int, execEndInfo]
+	baseEvents  *lru.Cache[int, *trace.Event]
+	deriveBase  deriveBase
+}
+
+// newProcessExecuteFailedGenerator initialize a new generator for the ProcessExecuteFailed event.
+func newProcessExecuteFailedGenerator() (*execFailedGenerator, error) {
```

Comment:
Why make this a singleton if it's not publicly accessible? Just construct the struct once outside the returned function and it will live in that scope.

Reply:
Hmm its just not how it works right now with other functions.
I had initializations from the calling function with other derived events but they were removed.
Anyways, its more comftable API and I don't see a problem with doing it like this.

Reply:
If so, then it would also be redundant in them as well (unless they require public global access, not the case here as the singleton isn't exported or used in another internal file).